### PR TITLE
Enhancements for new admin/users index

### PR DESCRIPTION
**Pull request by mlooney**
* adds .roles attribute to user presenter
* eager-loads user<->roles relation to avoid n+1 issue displaying roles
for a user

### CODE REVIEW

File: app/presenters/hyrax/admin/users_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+module Hyrax
+  module Admin
+    class UsersPresenter
+      # @return [Array] an array of Users & and array of roles
+      def users
+        @users ||= search
+      end
+
+      def roles
```
Comment (mark-dce) [inline app/presenters/hyrax/admin/users_presenter.rb:10]: Are we going to need a way to reset this if we add a new role?

**Review by mark-dce — APPROVED** [review]

File: app/presenters/hyrax/admin/users_presenter.rb (hunk 1 above)
Comment (mark-dce) [reply app/presenters/hyrax/admin/users_presenter.rb:10]: i.e. if you look at a user, add a role, add the user to that role, and come back to the user - will the original role list still be memoized without the new role?

Comment (mlooney) [reply app/presenters/hyrax/admin/users_presenter.rb:10]: yeah, that would be an issue, but this is c&p from hyrax, we could probably fix that.

**Comment by mlooney** [comment]
passed by reality

**Pull request closed without merging**